superset/db_engine_specs/base.py

```diff
@@ -959,3 +959,15 @@ def mutate_db_for_connection_test(database: "Database") -> None:
         :param database: instance to be mutated
         """
         return None
+
+    @staticmethod
+    def mutate_connection_args(
+        database: "Database", connect_args: Dict[str, Any]
+    ) -> None:
+        """
+        Some databases require passing additional non-standard parameters to database
+        connections, for example client certificates.
+
+        :param database: instance to be mutated
```

Comment:
Nit. Missing `connect_args` params.

Reply:
Wait, connect_args is an attribute of the database isn't it?

Reply:
Yes, or more specifically it's in `database.extra`, which is in JSON format. The reason it's being passed as a separate argument is that it's already been parsed out of `extra`. We could optionally inject this into `extra`, too, but that might be more confusing, as then we'd have to `json.parse`, mutate and `json.dumps` again.

Reply:
@mistercrunch I changed this so extras is parsed in `db_engine_specs`, so that we don't need to mutate the `Database` instance (this caused the mutations to get persisted in the db when saving the database).